### PR TITLE
Botany: Actually Random Chems and Increased Probabilities

### PR DESCRIPTION
**Pull request by Iamgoofball**
cutshff
ppwpzfg

I ended up with a Plump Helmet that grew pure Meth. It also teleported shit.

And some grapes with Smoke Powder, CLF3, and MSG. I was able to stick some cables in it and make power. It only grew in dark places, though.

Also some vines with Neurotoxin and Morphine. The vines hated the dark, and expelled nitrogen into the air.

I also upped probabilities on a lot of the potential trait manifests, so that things are a bit more fun.

I also added 10 random seeds to the Vendor to encourage use of the new system and to allow for more bug reports to come in early on about the random seed system so we can get them squashed sooner.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Holder.dm [hunk 1]
```diff
@@ -23,6 +23,15 @@ datum
 				for(var/path in paths)
 					var/datum/reagent/D = new path()
 					chemical_reagents_list[D.id] = D
+			if(!safe_reagents_list)
+				//Filtered list to prevent shit like xenobotany spitting out 200 bar drinks
+				var/paths = typesof(/datum/reagent) - /datum/reagent
+				safe_reagents_list = list()
+				for(var/path in paths)
+					var/datum/reagent/D = new path()
```
Comment (Zuhayr) [inline code/modules/reagents/Chemistry-Holder.dm]: Don't do this. You can use initial() on an uninstantiated var to grab the value that type has defined.

**Comment by FalseIncarnate** [comment]
Personally, I don't think there should be any random seeds in the vendor. Since this is a job available to day 0 accounts, giving them 10 random seeds at roundstart is a lot of potential power.

Making them go through cargo helps ensure that only the people who are aware of what they are doing get the random seeds, and prevents there being a massive power advantage at roundstart for botany. Having vines in the halls within minutes of roundstart would be devastating. Having vines that eat people in the halls in the same amount of time would be catastrophic.

**Comment by Iamgoofball** [comment]
I had to do an extreme amount of grinding to get the vine seed I mentioned to show up, somewhere around the 50th seed spawned mark. Vines are still rare occurrences from random seeds, so this shouldn't be a problem.

We can always revert the seed packets later on, but the new botany stuff is a rather big feature, and we do want as much of it getting used as possible, for now.

Tl;Dr, let's not preemptively nerf it just yet, let's wait to see if it actually needs a nerf.

File: code/modules/reagents/Chemistry-Holder.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Holder.dm]: Will do, thanks.

**Comment by ZomgPonies** [comment]
Our position is that we want to cut down on power creep in PRs. If that means starting things off a little slower and increasing their power as needed, so be it. Your PR already adds a lot of randomness to the new botany system which could arguably making it much more dangerous, and False brings up valid reasons why that could be bad very quickly.

Cut down on the initial number of packets to 2, and we'll increase it if we see the system is underpowered.

**Comment by Zuhayr** [comment]
Space vine event will give you carnivorous and vine genes immediately if you can wirecutter the plants. You don't need to grind random plant packets to get them.

**Comment by Iamgoofball** [comment]
Oh, didn't know that.

**Comment by Fox-McCloud** [comment]
With False on this one--it's kinda like how Virology pretty much has to go to cargo to get a decent virus selection---or really know what they're doing with their current dishes.

**Comment by ZomgPonies** [comment]
Better make the pack adjustments now if you want to get his merged today.

**Comment by Iamgoofball** [comment]
I can in about 30 minutes, encountered technical difficulty

**Comment by Iamgoofball** [comment]
okay so since byond is shitting itself severely, I can do github edits that are easy like the seed amount change, but I don't trust myself to make that inital thing work with a github edit, so either this gets merged and one of you guys does it in post, or we delay the PR and hope byond stops crying internally whenever I stare at it

**Comment by ZomgPonies** [comment]
DNM until you get your shit sorted, no stress.

File: code/modules/reagents/Chemistry-Holder.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Holder.dm]: okay so either I'm doing this horrifyingly wrong or this doesn't work but trying this results in "tried to pick from empty list"

i tried 3 different methods, all of them failed, i think this is the only way to go

**Comment by Mijhead** [comment]
There's actually already a vending machine in the code that has a bunch of random seeds in it, and it even lists particulary notable traits on them in the list (Such as spread, parasitism, carnivorism, etc.), with a red version of the name if it's tier 2, but it doesn't use the same vending machine UI as we normally have, instead it uses the bay one, which is all-white and stuff.

And a small suggestion I'd like to make, to make blue tomatoes slippery again, is to give them juicy 2 as a trait, and once it reaches bluespace tomatoes return it to juicy 1

File: code/modules/reagents/Chemistry-Holder.dm (hunk 1 above)
Comment (ZomgPonies) [inline code/modules/reagents/Chemistry-Holder.dm]: Actually, if you're going to make a list of all reagent datums, do it ONCE in global_lists.dm and then call that.

**Comment by ZomgPonies** [comment]
Soooooooooo is this still a thing or not?

**Comment by Iamgoofball** [comment]
I've had a lot of shit going on, this should be in working order though since the merge conflict was fixed.

**Comment by Iamgoofball** [comment]
wait what happened to surgery stuff what

**Comment by ZomgPonies** [comment]
Did you make the packet adjustment as requested or not?

**Comment by FalseIncarnate** [comment]
As of the time of this comment, the vendor has been unchanged from the 10 random seed packets.

While I'm skeptical of some of the probability adjustments, I'll see how those play out in live before deciding whether or not they are too much extra chance.

Also, you should add nanomachines to the list of chems that should NOT be part of the random chem proc or able to be grown in plants. Once they get finished being updated, they WILL borg people, and that's not the kind of thing that should randomly show up in a plant or whatever the random chem list might get used for in the future.

And maybe it's just me, but I don't see a difference between the old and new versions of code/_HELPERS/global_lists.dm aside from removing a blank line at the top of the file. What's that about?

**Comment by Iamgoofball** [comment]
@FalseIncarnate what? it should be set to 2 random seeds

it might of gotten reverted in that broken merge

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -976,7 +976,7 @@
 					/obj/item/seeds/poppyseed = 3,/obj/item/seeds/ambrosiavulgarisseed = 3,/obj/item/seeds/whitebeetseed = 3,/obj/item/seeds/watermelonseed = 3,/obj/item/seeds/limeseed = 3,
 					/obj/item/seeds/lemonseed = 3,/obj/item/seeds/orangeseed = 3,/obj/item/seeds/grassseed = 3,/obj/item/seeds/cocoapodseed = 3,
 					/obj/item/seeds/cabbageseed = 3,/obj/item/seeds/grapeseed = 3,/obj/item/seeds/pumpkinseed = 3,/obj/item/seeds/cherryseed = 3,/obj/item/seeds/plastiseed = 3,/obj/item/seeds/riceseed = 3,
-					/obj/item/seeds/tobaccoseed = 3, /obj/item/seeds/coffeeaseed = 3, /obj/item/seeds/teaasperaseed = 3)
+					/obj/item/seeds/tobaccoseed = 3, /obj/item/seeds/coffeeaseed = 3, /obj/item/seeds/teaasperaseed = 3, /obj/item/seeds/random = 10)
```
Comment (FalseIncarnate) [inline code/game/machinery/vending.dm]: Still 10, must have been accidentally reverted in one of your commits or you forgot to commit it. Shit happens.

Comment (Iamgoofball) [inline code/game/machinery/vending.dm]: :^(

will fix

**Comment by ZomgPonies** [comment]
Also merge conflict.

**Comment by Regen1** [comment]
20 days and counting! This is just like that other PR

**Comment by ZomgPonies** [comment]
Yeah enough is enough. Either fix it or close it, clock's ticking.

**Comment by ZomgPonies** [comment]
Travis failed.

**Comment by Iamgoofball** [comment]
aaaaaaaaaaaaaaaaaaaaaaaaa

**Comment by ZomgPonies** [comment]
So is this ready for review?

**Comment by Iamgoofball** [comment]
@ParadiseSS13/commit-access
After 9 year in development, hopefully it will have been worth the wait.

**Comment by ZomgPonies** [comment]
Open a new PR once this is ready for merge.
